psydac/linalg/identity.py

```diff
@@ -67,6 +67,9 @@ def dot( self, v, out=None ):
 
         return v
 
+    def transpose(self, conjugate=False):
+        return NotImplementedError()
```

Comment:
You should `raise` the exception instead of returning it:
```suggestion
    def transpose(self, conjugate=False):
        raise NotImplementedError()
```

Reply:
[07eaf6a](https://github.com/pyccel/psydac/pull/273/commits/07eaf6a9b76fcb517aea1ab4bb8290b6f99c0fc3)